invenio_search/api.py

```diff
@@ -67,9 +69,12 @@ def search(self, user_info=None, collection=None, enhance=True, **kwargs):
             for enhancer in query_enhancers():
                 query = enhancer(query, user_info=user_info,
                                  collection=collection)
-
-        for walker in search_walkers():
+        try:
+            walker = ElasticSearchNoKeywordsDSL()
             query = query.accept(walker)
+        except KeywordAppearance:
```

Comment:
I think would be better to call this exception `QueryHasKeywords`, since the exception is used as a message passing technique.